keps/sig-apps/3939-allow-replacement-when-fully-terminated/README.md

```diff
@@ -0,0 +1,877 @@
+- E2e tests are in Testgrid and linked in KEP
+- The feature flag enabled by default
+
+#### GA
+
+- Address reviews and bug reports from Beta users
+- Lock the `JobPodReplacementPolicy` feature-gate to true
+
+#### Deprecation
+
+- Remove `JobPodReplacementPolicy` feature-gate in GA+2.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+#### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: JobPodReplacementPolicy
+  - Components depending on the feature gate: kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
```

Comment:
For PRR we are particularly interested in ensuring that when the enhancement is not in use (none of the new fields of this enhancement are set on a pod) that the behavior of the system is unchanged.  My reading of the KEP is that you've handle this correctly; it is unchanged.  Can you state that as simply as possible in this section?

Reply:
+1 for the overall comment

Let's not describe what happens if you set a newly introduced field - let's focus only on the case when as a user I don't do nothing.

My understanding is that there are two changes:
- we always start reporting `terminating` pods in the status
- if podFailurePolicy is set, we start defaulting the pod-replacement-policy to Failed, which changes the way we're tracking terminating pods

The second is in theory backward-incompatible, but given the PodFailurePolicy is still in beta, we can probably leave with it [though we should dcoument it clearly]